src/EditorFeatures/Core/Implementation/SplitComment/SplitCommentCommandHandler.cs

```diff
@@ -117,6 +117,8 @@ private static bool MatchesCommentStart(ITextSnapshotLine line, string commentSt
             if (!enabled)
                 return false;
 
+            // We're synchronously blocking in the command handler.  So use the direct synchronous path to get a root to
+            // avoid undue blocking on teh threadpool getting around to this work.
```

Comment:
nit: small typo

```suggestion
            // avoid undue blocking on the threadpool getting around to this work.
```

Reply:
but then no one will recognize my shibboleth.

Reply:
This is a case where we should be calling `GetSyntaxRootAsync`. The comment isn't correct anyway:

* If this is currently on a thread pool thread (due to `GetOptionsAsync` yielding), it will block a thread pool thread for the synchronous call
* If this is currently on the UI thread, it will not block a thread pool thread for a call to `GetSyntaxRootAsync` because it is not running on a thread pool thread and the UI thread is blocked anyway

Reply:
true.  will update.